python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -152,24 +151,17 @@ async def get_chat_message_contents(
             settings.function_choice_behavior.auto_invoke_kernel_functions = False
             return await self._send_chat_request(settings)
 
+    @override
     async def get_streaming_chat_message_contents(
         self,
         chat_history: ChatHistory,
-        settings: OpenAIChatPromptExecutionSettings,
+        settings: "PromptExecutionSettings",
         **kwargs: Any,
-    ) -> AsyncGenerator[list[StreamingChatMessageContent | None], Any]:
-        """Executes a streaming chat completion request and returns the result.
-
-        Args:
-            chat_history (ChatHistory): The chat history to use for the chat completion.
-            settings (OpenAIChatPromptExecutionSettings | AzureChatPromptExecutionSettings): The settings to use
-                for the chat completion request.
-            kwargs (Dict[str, Any]): The optional arguments.
-
-        Yields:
-            List[StreamingChatMessageContent]: A stream of
-                StreamingChatMessageContent when using Azure.
-        """
+    ) -> AsyncGenerator[list[StreamingChatMessageContent], Any]:
+        if not isinstance(settings, OpenAIChatPromptExecutionSettings):
```

Comment:
nit: This check is probably not necessary as `get_prompt_execution_settings_from_settings` already has the logic implemented.

Reply:
I don't want to recreate something that is already valid, recreating prompt execution settings is also not a "cheap" operation since it packs and unpacks everything through extension_data, so prefer to first check and then only if not the right type proceed

Reply:
no, we don't want to recreate settings if not needed, hence this check first.

Reply:
Oh no, I mean `get_prompt_execution_settings_from_settings` will check if the settings is the right instance first.